dandischema/datacite.py

```diff
@@ -176,21 +176,39 @@ def to_datacite(
         attributes["relatedIdentifiers"] = []
         for rel_el in meta.relatedResource:
             if rel_el.identifier is not None:
-                if ":" not in rel_el.identifier:
-                    continue
-                ident_tp, ident_id = rel_el.identifier.split(":", 1)
-                if ident_tp.lower() in DATACITE_MAP:
-                    ident_tp = DATACITE_MAP[ident_tp.lower()]
-                else:
-                    raise ValueError(
-                        f"identifier has to be from the list: {DATACITE_IDENTYPE}, "
-                        f"but {ident_tp} provided"
-                    )
+                if rel_el.identifier.lower().startswith("doi"):
+                    ident_tp, ident_id = rel_el.identifier.split(":", 1)
+                    if ident_tp.lower() in DATACITE_MAP:
+                        ident_tp = DATACITE_MAP[ident_tp.lower()]
+                    else:
+                        raise ValueError(
+                            f"identifier has to be from the list: {DATACITE_IDENTYPE}, "
+                            f"but {ident_tp} provided"
+                        )
+                else:  # trying to find identifier if url provided
+                    if "doi.org/" in rel_el.identifier:
+                        ident_tp = "DOI"
+                        ident_id = rel_el.identifier.split("doi.org/")[1]
+                    elif "osf.io/" in rel_el.identifier:
```

Comment:
we may want to keep osf.io urls as urls. not all osf.io urls are DOIs. only some are. so turning them into DOIs would require actually checking if the DOI is valid.